### PR TITLE
Modify interaction

### PR DESCRIPTION
**Pull request by jsalankey**
Adds an adapted version of the Modify Interaction provided by Openlayers. This interaction allows for dynamically updating geometries on the map with clicking and dragging. The 3D sync of the interaction is accomplished through the Drawing Layer and new dynamic synchronizers rather than the traditional method used for other pointer interactions (which set update/cleanup functions). This pattern should be cleaner and generally more easily adaptable to other use cases.

There are a few little quirks I still need to iron out:
- The vertex feature gets stuck sometimes in 2D.
- The alt-click interaction is a little funky and doesn't seem to work sometimes when it should. Might be an OL issue.
- Interpolated geometries don't modify as expected due to the intermediate coordinates. This is an underlying issue that might not be fixable in this issue.

### CODE REVIEW

File: src/os/interaction/modifyinteraction.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    this.keyHandler.listen(KeyHandler.EventType.KEY, this.handleKeyEvent, true, this);
+
+    // jank alert: the functions that are called when the interaction starts and ends are hard to override, so instead
+    // listen to our own events and toggle the map movement on and off
+    ol.events.listen(this, ol.interaction.ModifyEventType.MODIFYSTART, this.handleStart, this);
+    ol.events.listen(this, ol.interaction.ModifyEventType.MODIFYEND, this.handleEnd, this);
+
+    // use the drawing layer as the overlay layer so that geometries are synced to WebGL automatically
+    this.overlay_ = os.map.mapContainer.getDrawingLayer();
```
Comment (glennkitchellcaci) [inline src/os/interaction/modifyinteraction.js:58]: Nit: missing require() and .getInstance()

File: src/os/ui/help/controlblock.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  /**
+   * Get the key text
+   * @param {string} other
+   * @return {?string}
+   * @export
+   */
+  getFont(other) {
+    var font = os.ui.help.Controls.FONT_CLASS[other];
```
Comment (glennkitchellcaci) [inline src/os/ui/help/controlblock.js:103]: Nit: require()

File: test/plugin/cesium/sync/dynamicmultipolygonconverter.test.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        [[0, 0], [0, 5], [5, 5], [5, 0], [0, 0]]
+      ],
+      [
+        [[10, 10], [10, 20], [20, 20], [20, 10], [10, 10]]
+      ]
+    ];
+    geometry = new ol.geom.MultiPolygon(coords);
+    feature = new os.feature.DynamicFeature(geometry);
+    style = new ol.style.Style();
+    layer = new os.layer.Vector();
```
Comment (glennkitchellcaci) [inline test/plugin/cesium/sync/dynamicmultipolygonconverter.test.js:35]: Nit: A couple more require()'s os.layer, os.feature

**Review by glennkitchellcaci — CHANGES_REQUESTED** [review]
Found a few os namespace things to require() in. Hopefully it doesn't create any circular dependencies.

File: src/os/interaction/modifyinteraction.js (hunk 1 above)
Comment (schmidtk) [reply src/os/interaction/modifyinteraction.js:58]: `ol.events` and `ol.interaction.ModifyEventType` should also be referenced by require.

Comment (jsalankey) [reply src/os/interaction/modifyinteraction.js:58]: I'll add a comment here. Requiring the `MapContainer` was creating a circular dependency somewhere, which was why I went to the sucky reference. I take a quick look to see if I can untangle it, but that's a big mess for later.

File: test/plugin/cesium/sync/dynamicmultipolygonconverter.test.js (hunk 3 above)
Comment (jsalankey) [reply test/plugin/cesium/sync/dynamicmultipolygonconverter.test.js:35]: These are all required correctly. The base OL classes and `DynamicFeature` are not modules yet and so I don't think they can be included with `goog.module.get`.

**Review by schmidtk — APPROVED** [review]

**Review by schmidtk — APPROVED** [review]

**Pull request merged**